addons/mail_client_extension/controllers/main.py

```diff
@@ -0,0 +1,323 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import datetime
+import hashlib
+import hmac
+import json
+import logging
+import odoo
+import werkzeug
+
+from odoo import http, tools
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class MailClientExtensionController(http.Controller):
+
+    @http.route('/mail_client_extension/auth', type='http', auth="user", methods=['GET'], website=True)
+    def auth(self, **values):
+        """
+         Once authenticated this route renders the view that shows an app wants to access Odoo.
+         The user is invited to allow or deny the app. The form posts to `/mail_client_extension/auth/confirm`.
+         """
+        return request.render('mail_client_extension.app_auth', values)
+
+    @http.route('/mail_client_extension/auth/confirm', type='http', auth="user", methods=['POST'])
+    def auth_confirm(self, scope, friendlyname, redirect, info=None, do=None, **kw):
+        """
+        Called by the `app_auth` template. If the user decided to allow the app to access Odoo, a temporary auth code
+        is generated and he is redirected to `redirect` with this code in the URL. It shoudl redirect to the app, and
+        the app should then exchange this auth code for an access token by calling
+        `/mail_client_extension/auth/access_token`.
+        """
+        params = {}
+        if do:
+            name = friendlyname if not info else friendlyname + ' ' + info
```

Comment:
Maybe add a colon to more clearly separate "friendlyname" and info? e.g.
```suggestion
            name = friendlyname if not info else f'{friendlyname}: {info}'
```